lib/client.js

```diff
@@ -794,9 +794,9 @@ class Parser {
   }
 
   onMessageComplete () {
-    const { client, socket, statusCode, upgrade, trailers, headers: rawTrailers } = this
+    const { client, socket, statusCode, upgrade, trailers, headers: rawTrailers, shouldKeepAlive } = this
 
-    if (socket.destroyed) {
+    if (socket.destroyed && (!statusCode || shouldKeepAlive)) {
```

Comment:
This might cause problems. I need to think a little. The current invariant is that no callbacks are invoked while `socket.destroyed`.

Reply:
This is needed as the socket is already destroyed due to the incoming error. I added the check on the statusCode and shouldKeepAlive instead of removing because otherwise https://github.com/nodejs/undici/blob/main/test/client-idempotent-body.js was failing.

Reply:
Yea I get that. I'll just give it some thought to ensure it doens't have any unforseen side effect.